ironic_prometheus_exporter/parsers/manager.py

```diff
@@ -0,0 +1,26 @@
+from ironic_prometheus_exporter.parsers import ipmi
+
+
+class ParserManager(object):
+
+    def __init__(self, data):
+
+        node_name = data['payload']['node_name']
+        payload = data['payload']['payload']
+        self.ipmi_objects = [
+            ipmi.Management(payload['Management'], node_name),
+            ipmi.Temperature(payload['Temperature'], node_name),
+            ipmi.System(payload['System'], node_name),
+            ipmi.Current(payload['Current'], node_name),
+            ipmi.Version(payload['Version'], node_name),
+            ipmi.Memory(payload['Memory'], node_name),
+            ipmi.Power(payload['Power'], node_name),
+            ipmi.Watchdog2(payload['Watchdog2'], node_name),
+            ipmi.Fan(payload['Fan'], node_name)
```

Comment:
Actually, all these classes should be just functions that you call sequentially.

Unless you want to go fancy and define setuptools entrypoints for each payload item. Even in this case they could be just functions.

Reply:
you mean change the classes in ipmi module to be functions  and we just call them?

Reply:
there are very few differences between the classes, I think @dtantsur is right and for example we can just have a list of sensors and call the different functions based on that with little tweaks based on the sensor type, for example if parsing requirements are different

Reply:
@dtantsur @elfosardo I did some changes let me know what you think =)

Reply:
Are we good with the changes?